test/e2e-go/features/stateproofs/stateproofs_test.go

```diff
@@ -173,3 +176,169 @@ func verifyStateProofForRound(r *require.Assertions, libgoal libgoal.Client, res
+			nextStateProofRound := uint64(lastStateProofBlock.Round()) + consensusParams.StateProofInterval
+
+			t.Logf("found a state proof for round %d at round %d", nextStateProofRound, blk.Round())
+			// Find the state proof transaction
+			stateProofMessage, nextStateProofBlock := verifyStateProofForRound(r, libgoal, restClient, nextStateProofRound, lastStateProofMessage, lastStateProofBlock, consensusParams)
+			lastStateProofMessage = stateProofMessage
+			lastStateProofBlock = nextStateProofBlock
+		}
+	}
+	r.Equalf(consensusParams.StateProofInterval*expectedNumberOfStateProofs, uint64(lastStateProofBlock.Round()), "the expected last state proof block wasn't the one that was observed")
+}
+
+func TestUnableToRecoverFromLaggingStateProofChain(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	r := require.New(fixtures.SynchronizedTest(t))
+
+	configurableConsensus := make(config.ConsensusProtocols)
+	consensusVersion := protocol.ConsensusVersion("test-fast-stateproofs")
+	consensusParams := config.Consensus[protocol.ConsensusCurrentVersion]
+	consensusParams.StateProofInterval = 16
+	consensusParams.StateProofTopVoters = 1024
+	consensusParams.StateProofVotersLookback = 2
+	// Stateproof can be generated even if not all nodes function correctly. e.g node can be offline
+	// and stateproofs might still get generated. in order to make sure that all nodes work correctly
+	// we want the network to fail in generating stateproof if one node is not working correctly.
+	// For that we will increase the proven Weight to be close to 100%. However, this change might not be enough.
+	// if the signed Weight and the Proven Weight are very close to each other the number of reveals in the state proof
+	// will exceed the MAX_NUMBER_OF_REVEALS and proofs would not get generated
+	// for that reason we need to the decrease the StateProofStrengthTarget creating a "weak cert"
+	consensusParams.StateProofWeightThreshold = (1 << 32) * 90 / 100
+	consensusParams.StateProofStrengthTarget = 4
+	consensusParams.StateProofRecoveryInterval = 3
+	consensusParams.EnableStateProofKeyregCheck = true
+	consensusParams.AgreementFilterTimeout = 1500 * time.Millisecond
+	consensusParams.AgreementFilterTimeoutPeriod0 = 1500 * time.Millisecond
+	configurableConsensus[consensusVersion] = consensusParams
+
+	var fixture fixtures.RestClientFixture
+	fixture.SetConsensus(configurableConsensus)
+	fixture.Setup(t, filepath.Join("nettemplates", "StateProof.json"))
+	defer fixture.Shutdown()
+
+	err := fixture.WaitForRound(1, 30*time.Second)
+	r.NoError(err)
+
+	dir, err := fixture.GetNodeDir("Node4")
+	nc := nodecontrol.MakeNodeController(fixture.GetBinDir(), dir)
+	nc.FullStop()
+
+	var lastStateProofBlock bookkeeping.Block
+
+	libgoal := fixture.LibGoalClient
+	for rnd := uint64(2); rnd <= consensusParams.StateProofInterval*(expectedNumberOfStateProofs+1); rnd++ {
```

Comment:
Give the network more time after restarting the node, and make sure the network continues to make progress. 
```suggestion
	for rnd := uint64(2); rnd <= consensusParams.StateProofInterval*(consensusParams.StateProofRecoveryInterval+3); rnd++ {
```